src/analysis/complex/isometry.lean

```diff
@@ -74,77 +87,32 @@ begin
   apply linear_isometry.im_apply_eq_im h,
 end
 
-lemma linear_isometry_complex_aux (f : ℂ →ₗᵢ[ℝ] ℂ) (h : f 1 = 1) :
-  (∀ z, f z = z) ∨ (∀ z, f z = conj z) :=
+lemma linear_isometry_complex_aux {f : ℂ ≃ₗᵢ[ℝ] ℂ} (h : f 1 = 1) :
+  f = linear_isometry_equiv.refl ℝ ℂ ∨ f = conj_lie :=
 begin
   have h0 : f I = I ∨ f I = -I,
-  { have : |f I| = 1,
-    { rw [←norm_eq_abs, linear_isometry.norm_map, norm_eq_abs, abs_I] },
+  { have : |f I| = 1 := by simpa using f.norm_map complex.I,
     simp only [ext_iff, ←and_or_distrib_left, neg_re, I_re, neg_im, neg_zero],
     split,
     { rw ←I_re,
-      rw linear_isometry.re_apply_eq_re h },
-    { apply linear_isometry.im_apply_eq_im_or_neg_of_re_apply_eq_re,
-      { intro z, rw [←norm_eq_abs, ←norm_eq_abs, linear_isometry.norm_map] },
-      { intro z, rw linear_isometry.re_apply_eq_re h } } },
-  refine or.imp (λ h1, _) (λ h1 z, _) h0,
-  { suffices : f.to_linear_map = linear_isometry.id.to_linear_map,
-    { simp [this, ←linear_isometry.coe_to_linear_map, linear_map.id_apply] },
-    apply basis.ext basis_one_I,
-    intro i,
-    fin_cases i,
-    { simp [h] },
-    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
-        linear_map.id_coe, id.def, matrix.cons_val_one], simpa } },
-  { suffices : f.to_linear_map = conj_lie.to_linear_equiv,
-    { rw [←linear_isometry.coe_to_linear_map, this],
-      simp only [linear_isometry.coe_to_linear_map], refl },
-    apply basis.ext basis_one_I,
-    intro i,
-    fin_cases i,
-    { simp only [h, linear_isometry.coe_to_linear_map, matrix.cons_val_zero], simpa },
-    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
-        linear_map.id_coe, id.def, matrix.cons_val_one], simpa } },
+      exact @linear_isometry.re_apply_eq_re f.to_linear_isometry h I, },
+    { apply @linear_isometry.im_apply_eq_im_or_neg_of_re_apply_eq_re f.to_linear_isometry,
+      intro z, rw @linear_isometry.re_apply_eq_re f.to_linear_isometry h } },
+  refine h0.imp (λ h' : f I = I, _) (λ h' : f I = -I, _);
+  { apply linear_isometry_equiv.to_linear_equiv_injective,
+    apply complex.basis_one_I.ext',
+    intros i,
+    fin_cases i; simp [h, h'] }
 end
 
-lemma linear_isometry_complex (f : ℂ →ₗᵢ[ℝ] ℂ) :
-  ∃ a : ℂ, |a| = 1 ∧ ((∀ z, f z = a * z) ∨ (∀ z, f z = a * conj z)) :=
+lemma linear_isometry_complex (f : ℂ ≃ₗᵢ[ℝ] ℂ) :
```

Comment:
The old version seems slightly stronger, as it only assumes a linear isometric map instead of a linear isometric equiv. Of course, this is equivalent as we are in finite dimension. Do you think you could restate the theorem for linear isometric maps?

Reply:
For me, the statement with equivs seems more elegant, and perhaps more useful in applications, because one would always in practice work with equivs, and to deduce the equality of two equivs from the equality of their underlying maps requires wielding the lemma `linear_isometry_equiv.to_linear_equiv_injective`.

What do you think of instead adding a lemma in `analysis.normed_space.linear_isometry` that in finite dimension linear-isometry-maps are linear-isometry-equivs?  (Then mentioning that lemma in the docstring of this file.)

I don't feel very strongly, so let me know if you prefer your original proposal.

Reply:
Your proposal sounds good. And it can be made in another PR, so let's merge this one.